code/modules/admin/quantum_mechanic.dm

```diff
@@ -0,0 +1,323 @@
+	id_slot =  slot_wear_id_str
+
+/mob/living/carbon/human/quantum
+	status_flags = NO_ANTAG
+	universal_understand = TRUE
+	var/fall_override = TRUE
+	movement_handlers = list(
+		/datum/movement_handler/mob/relayed_movement,
+		/datum/movement_handler/mob/death,
+		/datum/movement_handler/mob/conscious,
+		/datum/movement_handler/mob/eye,
+		/datum/movement_handler/move_relay,
+		/datum/movement_handler/mob/buckle_relay,
+		/datum/movement_handler/mob/delay,
+		/datum/movement_handler/mob/stop_effect,
+		/datum/movement_handler/mob/physically_capable,
+		/datum/movement_handler/mob/physically_restrained,
+		/datum/movement_handler/mob/space,
+		/datum/movement_handler/mob/multiz,
+		/datum/movement_handler/mob/multiz_connected,
+		/datum/movement_handler/mob/movement
+	)
+
+/mob/living/carbon/human/quantum/can_inject(mob/user, target_zone)
+	to_chat(user, SPAN_DANGER("\The [src] disarms you before you can inject them."))
+	user.drop_item()
+	return FALSE
+
+/mob/living/carbon/human/quantum/binarycheck()
+	return TRUE
+
+/mob/living/carbon/human/quantum/proc/suicide()
+	if(QDELETED(src))
+		return
+
+	custom_emote(VISIBLE_MESSAGE, "presses a button on their suit, followed by a polite bow.")
+	spark_at(src)
+
+	QDEL_IN(src, 1 SECOND)
+
+	if(key)
+		var/mob/observer/ghost/ghost = new(src)	//Transfer safety to observer spawning proc.
+		ghost.key = key
+		ghost.dir = dir
+		ghost.mind.name = "[ghost.key] Quantum Mechanic"
+		ghost.name = "[ghost.key] Quantum Mechanic"
+		ghost.real_name = "[ghost.key] Quantum Mechanic"
+		ghost.voice_name = "[ghost.key] Quantum Mechanic"
+		ghost.admin_ghosted = TRUE
+		ghost.can_reenter_corpse = TRUE
+		ghost.reload_fullscreen()
+
+/mob/living/carbon/human/quantum/verb/quantum_antigrav()
+	set name = "Toggle Gravity"
+	set desc = "Toggles on/off falling for you."
+	set category = "Ω"
+
+	if (fall_override)
+		fall_override = FALSE
+		to_chat(usr, SPAN_NOTICE("You will now fall normally."))
+	else
+		fall_override = TRUE
+		to_chat(usr, SPAN_NOTICE("You will no longer fall."))
+
+/mob/living/carbon/human/quantum/verb/quantum_walk()
+	set name = "Ruin Everything"
+	set desc = "Uses bluespace technology to phase through solid matter and move quickly."
+	set category = "Ω"
+	set popup_menu = 0
+
+	if(usr.HasMovementHandler(/datum/movement_handler/mob/incorporeal))
+		usr.RemoveMovementHandler(/datum/movement_handler/mob/incorporeal)
+		to_chat(usr, SPAN_NOTICE("You will no-longer phase through solid matter."))
+	else
+		usr.ReplaceMovementHandler(/datum/movement_handler/mob/incorporeal)
+		to_chat(usr, SPAN_NOTICE("You will now phase through solid matter."))
+
+/mob/living/carbon/human/quantum/verb/quantum_recover()
+	set name = "Rejuv"
+	set desc = "Use the quantum within you to restore your health."
+	set category = "Ω"
+	set popup_menu = FALSE
+
+	revive()
+
+/mob/living/carbon/human/quantum/verb/quantum_awake()
+	set name = "Wake up"
+	set desc = "This is a quick fix to the relogging sleep bug."
```

Comment:
What relogging sleep bug?

Reply:
We already have a Sleep verb that will wake you up if you're asleep, so this is unneeded.